src/core/drive/visit.js

```diff
@@ -159,7 +159,7 @@ export class Visit {
       this.simulateRequest()
     } else if (this.shouldIssueRequest() && !this.request) {
       this.request = new FetchRequest(this, FetchMethod.get, this.location)
-      this.request.perform()
+      await this.request.perform()
```

Comment:
Wouldn't this require that `issueRequest` be asynchronous? And then the entire preceding calls as well?